### DISCUSSION TITLE
WebGPU vs OpenGL

### DISCUSSION THREAD

**Discussion by ThanosFisherman** [littlektframework/littlekt#243] [Q&A]
Hey! Big fan of your work! I was wondering what are the benefits of migrating to WebGPU from OpenGL? Is it really worth it for a 2D framework?

**Comment by LeHaine** [comment]
Hey! There probably aren't many benefits other than that WebGPU being a modern API compared to OpenGL / WebGL. Not that OpenGL is deprecated or obsolete, it just felt right to have it migrate to something modern.

I have found, in my experience of migrating LittleKt over to WebGPU, the API to feel more intuitive and made more sense to me than OpenGL. But, of course, some may find the opposite true.

**Reply by ThanosFisherman** [reply]
Many people say that webgpu feels more intuitive than OpenGL despite being more verbose I wouldn't know personally. Are you planning to keep OpenGL alongside webgpu and have the user choose or are you going full webgpu in the future?

**Reply by LeHaine** [reply]
I am not entirely sure yet. With the WebGPUupdate, I am making the switch to use the package name `com.littlekt.*` instead of the `com.lehaine.littlekt` package. So with that we could continue having the OpenGL be the same but that would also mean double the work 😃

**Reply by ThanosFisherman** [reply]
I mean if both of the APIs serve the same purpose in terms of portability (especially on mobile) and performance, it'd make sense to go full webgpu although I assume it'd be a bit cumbersome to maintain the API since it's not included in lwjgl and from what I see you did a lot of binding work yourself. Anyhow whatever you decide I'm interested to see how it works out. 🙂
